arelle/Cntlr.py

```diff
@@ -8,18 +8,26 @@
    :license: Apache-2.
    :synopsis: Common controller class to initialize for platform and setup common logger functions
 """
-from arelle import PythonUtil # define 2.x or 3.x string types
+from __future__ import annotations
+from typing import Any, TYPE_CHECKING, TextIO, Mapping
+from .typing import TypeGetText
 import tempfile, os, io, sys, logging, gettext, json, re, subprocess, math
 from arelle import ModelManager
+from arelle.WebCache import WebCache
 from arelle.Locale import getLanguageCodes, setDisableRTL
 from arelle import PluginManager, PackageManager
 from collections import defaultdict
+
+_: TypeGetText
```

Comment:
I don't quite follow what this does

Reply:
I introduced this in another PR. `_` seems to be shorthand for translating a string using gettext so this line adds a type hint stating that `_` is a callable that takes a `str` as an argument and returns a `str` (or `Any` as discussed elsewhere in this PR).